hnn_core/parallel_backends.py

```diff
@@ -175,6 +190,13 @@ def run_subprocess(command, obj, timeout, *args, **kwargs):
                 # both exit conditions have been met (also we know that
                 # the network has been sent)
                 break
+
+            if not child_terminated and \
+                    count_since_last_output > timeout_cycles:
+                warn("No output from child process in approx 10s. "
+                     "Terminating...")
+                kill_proc_name('nrniv')
```

Comment:
same comment. Or are you trying to run multiple `simulate_dipole` in one backend? Is there a usecase for that? You could do for example:

```py
for _ in range(10):
    with MPIBackend(...):
        simulate_dipoles(...)
```

and this should work without this line as long as you have `kill_proc_name` in the `__exit__`

Reply:
No, not trying to do that. See above. Also, `kill_proc_name` will not do anything if there are no `nrniv` processes running.